docs/src/pages/guides/routing/ButtonDemo.js

```diff
@@ -1,15 +1,16 @@
 import * as React from 'react';
-import Box from '@material-ui/core/Box';
 import Button from '@material-ui/core/Button';
 
-const preventDefault = (event) => event.preventDefault();
+function preventDefault(event) {
+  event.preventDefault();
+}
 
 export default function ButtonDemo() {
   return (
-    <Box role="presentation" onClick={preventDefault}>
+    <div role="presentation" onClick={preventDefault}>
```

Comment:
Let's just remove this. People do want to see our components in action.

Reply:
@oliviertassinari should I do these changes?

Reply:
I would personally vote for keeping the prevent default for the same argument used by @eps1lon. I think that developers want to see what's the UX when the links are clicked. Is there a ripple? a `:visited` style? But don't want to break the flow of their navigation in the docs. It also wouldn't be consistent with https://next.material-ui.com/components/links/#basic-links. However, I have applied the change, so we can experiment: db059ca649. 

In our case, the docs intercept the clicks on the native `<a>` element, to trigger Next.js's router.

For the other sections, I strongly think that not breaking the flow of reading the docs is more important.

Reply:
> For the other sections, I strongly think that not breaking the flow of reading the docs is more important.

The question I would ask is what the expectation is when clicking the link. They can always "just revert" navigation. That's how the browser works.

Or do we want to optimize for an accidental click by developer and "breaking of reading flow". I don't think we actually break that reading flow when people click. That's an intentional action.

Reply:
My assumption is that the main thing a developer will care about on the button is how the UI handles the interaction, not the functional of the navigation itself. For instance, Next.js didn't even bother to demonstrate it in its link documentation page.